io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -312,7 +279,18 @@ public WfsBackedGmlInstanceIterator iterator() {
 	 */
 	@Override
 	public boolean isEmpty() {
-		return size == 0;
+		if (!emptyInitialized) {
+			ResourceIterator<Instance> it = iterator();
```

Comment:
I think a more effective way could be using the hits query that was previously used - it could even be used with `maxFeatures` / `count` set to `1` as we are only interested if there are matches at all.
With that we would avoid the server having to load and transfer a potentially large amount of data.